Migrate nose tests to pytest

ethho opened the pull request:
As scoped in [PLAT-107](https://datajoint.atlassian.net/browse/PLAT-107), migrate the nose tests in `tests_old` to pytest-compatible tests in the `tests` directory. Since nose is not supported after Python v3.8, this is necessary if we want `datajoint-python` to support modern versions of Python.

In addition, fix and update the [devcontainer](https://code.visualstudio.com/docs/devcontainers/containers) development environment.

### Component PRs

This PR contains changes made in the following PRs:

- #1114
- #1115
- #1116
- #1117
- #1118
- #1119
- #1120
- #1121
- #1122
- #1123
- #1124
- #1125
- #1126
- #1127
- #1128
- #1129
- #1130
- #1131
- #1132
- #1133
- #1134
- #1135
- #1136
- #1137
- #1138
- #1139
- #1140
- #1141
- #1143 
- #1144
- #1146
- #1147

A-Baji commented:
@ethho I deleted the converted modules from `tests_old` in the [PR](https://github.com/datajoint/datajoint-python/pull/1114) for the first pass of the conversion. We probably want to revert that for archiving purposes.

ethho commented:
> @ethho I deleted the converted modules from `tests_old` in the [PR](https://github.com/datajoint/datajoint-python/pull/1114) for the first pass of the conversion. We probably want to revert that for archiving purposes.

I added these removed nosetests on [00c666a](https://github.com/datajoint/datajoint-python/pull/1142/commits/00c666adc540f7b4857ff0fe44aa49818369fd7b)
